easybuild/tools/run.py

```diff
@@ -244,6 +244,9 @@ def to_cmd_str(cmd):
     if output_file:
         toptmpdir = os.path.join(tempfile.gettempdir(), 'run-shell-cmd-output')
         os.makedirs(toptmpdir, exist_ok=True)
+        # restrict the allowed characters in the name of the output_file
+        allowed_chars = string.ascii_letters + string.digits
```

Comment:
@branfosj We could also allow `-` and `_`? Not sure that would make much of a difference in practice though...

We should also have a test to verify that this indeed does what we expect, let me know if you need help with this.

I'm fine with enhancing the existing `test_run_shell_cmd_basic` in `test/framework/run.py` for this.

Reply:
see faf8f14

It was eaier for me to just pull that code out into a separate function and test that directly.